packages/viewer/src/components/viewer-markup-arrow/viewer-markup-arrow.tsx

```diff
@@ -0,0 +1,352 @@
+
+  @Element()
+  private hostEl!: HTMLElement;
+
+  @State()
+  private elementBounds?: DOMRect;
+
+  @State()
+  private editAnchor: ViewerMarkupArrowEditAnchor = 'end';
+
+  /**
+   * @ignore
+   */
+  protected componentWillLoad(): void {
+    this.updateViewport();
+
+    this.handleViewerChanged(this.viewer);
+
+    this.updatePointsFromProps();
+  }
+
+  protected componentDidLoad(): void {
+    this.updatePointsFromProps();
+
+    const resize = new ResizeObserver(() => this.updateViewport());
+    resize.observe(this.hostEl);
+  }
+
+  @Method()
+  public async dispose(): Promise<void> {
+    if (this.viewer != null) {
+      this.removeInteractionListeners(this.viewer);
+    }
+    this.removeDrawingInteractionListeners();
+  }
+
+  /**
+   * @ignore
+   */
+  @Watch('viewer')
+  protected handleViewerChanged(
+    newViewer?: HTMLVertexViewerElement,
+    oldViewer?: HTMLVertexViewerElement
+  ): void {
+    if (oldViewer != null) {
+      this.removeInteractionListeners(oldViewer);
+    }
+
+    if (newViewer != null) {
+      this.addInteractionListeners(newViewer);
+    }
+  }
+
+  @Watch('start')
+  protected handleStartJsonChange(): void {
+    this.updatePointsFromProps();
+  }
+
+  @Watch('end')
+  protected handleEndJsonChange(): void {
+    this.updatePointsFromProps();
+  }
+
+  private updateViewport(): void {
+    const rect = getMarkupBoundingClientRect(this.hostEl);
+    this.elementBounds = rect;
+  }
+
+  private updatePointsFromProps(): void {
+    this.start = this.start || parsePoint(this.startJson);
+    this.end = this.end || parsePoint(this.endJson);
+  }
+
+  public render(): h.JSX.IntrinsicElements {
+    if (this.start != null && this.end != null && this.elementBounds != null) {
+      const screenStart = translatePointToScreen(
+        this.start,
+        this.elementBounds
+      );
+      const screenEnd = translatePointToScreen(this.end, this.elementBounds);
+      const arrowheadPoints = createArrowheadPoints(screenStart, screenEnd);
+
+      return (
+        <Host>
+          <svg class="svg">
+            <defs>
+              <SvgShadow id="arrow-shadow" />
+            </defs>
+            <g filter="url(#arrow-shadow)">
+              <polygon
+                id="arrow-head"
+                class="head"
+                points={arrowheadPointsToPolygonPoints(arrowheadPoints)}
+              />
+              <line
+                id="arrow-line"
+                class="line"
+                x1={screenStart.x}
+                y1={screenStart.y}
+                x2={arrowheadPoints.base.x}
+                y2={arrowheadPoints.base.y}
+              />
+              {this.mode === 'edit' && (
+                <line
+                  id="bounding-box-1d-line"
+                  class="bounds-line"
+                  x1={screenStart.x}
+                  y1={screenStart.y}
+                  x2={screenEnd.x}
+                  y2={screenEnd.y}
+                />
+              )}
```

Comment:
@danschultz converted everything bounding box related to HTML over SVG except for this `<line />` element. Didn't have much luck consistently centering that on the arrow with a div. Had some difficulty with the `rotateZ` and positioning it with `translate`. Open to suggestions here, but since this isn't possible to interact with, I left it as an SVG for now